duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/ui/SearchInterstitialActivity.kt

```diff
@@ -47,22 +49,47 @@ class SearchInterstitialActivity : DuckDuckGoActivity() {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
 
+        val params = intent.getActivityParams(SearchInterstitialActivityParams::class.java)
+        params?.query?.let { query ->
+            binding.duckChatOmnibar.duckChatInput.setText(query)
+        }
+
+        onBackPressedDispatcher.addCallback(
+            this,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    binding.duckChatOmnibar.animateOmnibarFocusedState(false)
+                    supportFinishAfterTransition()
+                }
+            },
+        )
+
         binding.duckChatOmnibar.apply {
             selectTab(0)
             enableFireButton = false
             enableNewChatButton = false
             onSearchSent = { query ->
-                startActivity(appBrowserNav.openInCurrentTab(context, query))
-                finish()
+                val data = Intent().putExtra(QUERY, query)
+                setResult(Activity.RESULT_OK, data)
+                supportFinishAfterTransition()
             }
             onDuckChatSent = { query ->
                 duckChat.openDuckChatWithAutoPrompt(query)
                 finish()
             }
-            onBack = { onBackPressed() }
+            onBack = {
+                val query = duckChatInput.text.toString()
+                val data = Intent().putExtra(QUERY, query)
+                setResult(Activity.RESULT_CANCELED, data)
+                onBackPressed()
+            }
         }
         binding.duckChatOmnibar.duckChatInput.post {
             showKeyboard(binding.duckChatOmnibar.duckChatInput)
         }
     }
+
+    companion object {
+        const val QUERY = "query"
```

Comment:
In the future once we know how this works we need to take a look at this `QUERY` API, it's in an impl module and accessed directly form the `:app` module. It also leaks a lot of implementation details, ie. caller needs to know there's a QUERY param, the type etc.